mozjs/js/rust/magic_codegen/src/lib.rs

```diff
@@ -55,6 +71,16 @@ fn match_struct(ast: &syn::DeriveInput, variant: &syn::VariantData) -> quote::To
             #(#struct_def,)*
         }
 
+        pub static #js_class : jsapi::JSClass = jsapi::JSClass {
+            name: #name_field as *const u8 as *const libc::c_char,
+            flags: (#num_reserved_slots &
+                    jsapi::JSCLASS_RESERVED_SLOTS_MASK) << jsapi::JSCLASS_RESERVED_SLOTS_SHIFT |
+            jsapi::JSCLASS_HAS_PRIVATE  |
+            #dom_flag,
```

Comment:
If the name doesn't begin with `DOM`, then this is going to be an empty identifier, which will result in a syntax error right here, correct?